include/material_models/J2Plasticity.h

```diff
@@ -0,0 +1,278 @@
+#ifndef J2PLASTICITY_H
+#define J2PLASTICITY_H
+
+#include "matmodel.h"
+#include "solver.h"
+
+class J2Plasticity : public MechModel {
+  public:
+    J2Plasticity(vector<double> l_e, map<string, vector<double>> materialProperties)
+        : MechModel(l_e)
+    {
+        try {
+            bulk_modulus  = materialProperties["bulk_modulus"];
+            shear_modulus = materialProperties["shear_modulus"];
+            yield_stress  = materialProperties["yield_stress"];                  // Initial yield stress
+            K             = materialProperties["isotropic_hardening_parameter"]; // Isotropic hardening parameter
+            H             = materialProperties["kinematic_hardening_parameter"]; // Kinematic hardening parameter
+            eta           = materialProperties["viscosity"];                     // Viscosity parameter
+            dt            = materialProperties["time_step"][0];                  // Time step
```

Comment:
What if the time step width is not constant over the load path?

Reply:
Unfortunately, we will support only fixed-time steps for now... not because it is hard algorithmically... but the input file will get more involved other-wise